frontends/web/src/contexts/WCWeb3WalletProvider.tsx

```diff
@@ -27,56 +28,69 @@ type TProps = {
 export const WCWeb3WalletProvider = ({ children }: TProps) => {
   const [web3wallet, setWeb3wallet] = useState<IWeb3Wallet>();
   const [isWalletInitialized, setIsWalletInitialized] = useState(false);
+  const config = useLoad(getConfig);
+  const hasUsedWC = config && config.frontend && config.frontend.hasUsedWalletConnect;
 
-  useEffect(() => {
-    if (!isWalletInitialized) {
-      const initializeWeb3Wallet = async () => {
-        try {
-          const core = new Core({
-            projectId: '89733df088867a1a1bf644013addd6cc',
-          });
+  const initializeWeb3Wallet = async () => {
+    try {
+      const { Core } = await import('@walletconnect/core');
+      const { Web3Wallet } = await import('@walletconnect/web3wallet');
 
-          const wallet = await Web3Wallet.init({
-            core,
-            metadata: {
-              name: 'BitBoxApp',
-              description: 'BitBoxApp',
-              url: 'https://bitbox.swiss',
-              icons: ['https://bitbox.swiss/assets/images/logos/dbb-logo.png']
-            }
-          });
+      const core = new Core({
+        projectId: '89733df088867a1a1bf644013addd6cc',
+      });
 
-          setWeb3wallet(wallet);
-          setIsWalletInitialized(true);
-        } catch (err: unknown) {
-          console.log('Error for initializing', err);
+      const wallet = await Web3Wallet.init({
+        core,
+        metadata: {
+          name: 'BitBoxApp',
+          description: 'BitBoxApp',
+          url: 'https://bitbox.swiss',
+          icons: ['https://bitbox.swiss/assets/images/logos/dbb-logo.png']
```

Comment:
@Tomasvrba what should we use for name and description? and isn't this also configured somewhere in walletconnect account?

I am currently thinking it should be 'BitBox' and the description should maybe mention that BitBox is a hardware wallet.

Also should this be translated or hardcoded to english language?

Reply:
BitBox as name is fine imo
and maybe something like "BitBox02 hardware wallet" as description

The thing in WC cloud is what is displayed in the explorer/when users selects a wallet to connect
The info entered here is what will be shown in the dapp during/after pairing

English is fine imo since there isn't really much text there

Reply:
ack makes sense to me.

Reply:
@shonsirsha feel free to change to "BitBox" and "BitBox02 hardware wallet" in this PR, or we can also do later in a different PR.. up to you.